pravega/pravega
# Issue 4274: Write padding more efficiently

**Change log description**  
* Write padding more efficiently instead of one Long at a time.

**Purpose of the change**  
Fixes #4274 

**What the code does**  
Encodes padding directly in CommandEncoder by using the netty API to write zeros to a byteBuff.

**How to verify it**  
Existing tests should cover this well, but to verify I walked through them in a debugger and discovered several were accidently broken in #3970 because they were passing 'null' for the ctx, which actually disabled batching. This was the cause of some of the changed assertion values in #3970. The tests now pass ctx properly and run with batching. The original assertions have been restored.
